packages/core/src/virtual_dom.rs

```diff
@@ -612,24 +602,29 @@ impl VirtualDom {
     /// ```
     #[instrument(skip(self, to), level = "trace", name = "VirtualDom::rebuild")]
     pub fn rebuild(&mut self, to: &mut impl WriteMutations) {
-        self.flush_templates(to);
+        self.maybe_rebuild(Some(to));
+    }
+
+    /// Rebuild the virtual dom without handling any of the mutations
+    pub fn maybe_rebuild<M: WriteMutations>(&mut self, to: Option<&mut M>) {
```

Comment:
It might make sense to make the virtual dom itself generic over mutations at some point? Then you could remove the generic on rebuild_in_place